gdal/gcore/gdal_priv.h

```diff
@@ -70,6 +70,7 @@ class GDALAsyncReader;
 #include <limits>
 #include <cmath>
 #include <memory>
+#include <iterator>
```

Comment:
Nit: Sort the C++ includes.  iterator before memory.

Reply:
I was tempted to sort them, but didn't know if there was some reason for this order. Why should iterator be before memory but not before limits or vector? Why is cmath after limits and limits after vector?

Reply:
We don't have an enforced style for gdal.  So this is my personal opinion.

https://trac.osgeo.org/gdal/wiki/rfc69_cplusplus_formatting stalled.

I try to get things moving towards what clang-format and iwyu want to do.  So C headers followed by C++ headers followed by other headers with each group alphabetized.  Much of GDAL doesn't follow that.  And here I was only considering your change.  iterator going before memory is more sorted.  If you want to sort more, I'd appreciate it, but I'm not going to ask you to do it.  Hopefully that makes sense.

Reply:
> https://trac.osgeo.org/gdal/wiki/rfc69_cplusplus_formatting stalled.

@schwehr Re-looking at the discussion, it seems to have not been pursued further, but there was no objection to the effort as far as I remember. I see there was some discussion about astyle vs clang-tidy, but if you want to pursue it, I don't see why this wouldn't come to a positive outcome

Reply:
Sure, and I think it is a good idea.